extensions/control-plane/api/management-api/secret-api/src/main/java/org/eclipse/edc/connector/api/management/secret/SecretApi.java

```diff
@@ -0,0 +1,148 @@
+            ),
+            responses = {
+                    @ApiResponse(responseCode = "200", description = "The secrets matching the query",
+                            content = @Content(array = @ArraySchema(schema = @Schema(implementation = SecretOutputSchema.class)))),
+                    @ApiResponse(responseCode = "400", description = "Request body was malformed",
+                            content = @Content(array = @ArraySchema(schema = @Schema(implementation = ApiCoreSchema.ApiErrorDetailSchema.class))))
+            })
+    JsonArray requestSecrets(JsonObject querySpecJson);
+
+    @Operation(description = "Gets a secret with the given ID",
+            responses = {
+                    @ApiResponse(responseCode = "200", description = "The secret",
+                            content = @Content(schema = @Schema(implementation = SecretOutputSchema.class))),
+                    @ApiResponse(responseCode = "400", description = "Request was malformed, e.g. id was null",
+                            content = @Content(array = @ArraySchema(schema = @Schema(implementation = ApiCoreSchema.ApiErrorDetailSchema.class)))),
+                    @ApiResponse(responseCode = "404", description = "A secret with the given ID does not exist",
+                            content = @Content(array = @ArraySchema(schema = @Schema(implementation = ApiCoreSchema.ApiErrorDetailSchema.class))))
+            }
+    )
+    JsonObject getSecret(String id);
+
+    @Operation(description = "Removes a secret with the given ID if possible. Deleting a secret is only possible if that secret is not yet referenced " +
+            "by a contract agreement, in which case an error is returned. " +
+            "DANGER ZONE: Note that deleting secrets referenced by an asset may lead to unexpected behavior in the system.",
+            responses = {
+                    @ApiResponse(responseCode = "204", description = "Secret was deleted successfully"),
+                    @ApiResponse(responseCode = "400", description = "Request was malformed, e.g. id was null",
+                            content = @Content(array = @ArraySchema(schema = @Schema(implementation = ApiCoreSchema.ApiErrorDetailSchema.class)))),
+                    @ApiResponse(responseCode = "404", description = "A secret with the given ID does not exist",
+                            content = @Content(array = @ArraySchema(schema = @Schema(implementation = ApiCoreSchema.ApiErrorDetailSchema.class)))),
+                    // TODO: check if it makes sense to check id secret is referenced from any asset
+                    @ApiResponse(responseCode = "409", description = "The secret cannot be deleted, because it is referenced by an asset",
+                            content = @Content(array = @ArraySchema(schema = @Schema(implementation = ApiCoreSchema.ApiErrorDetailSchema.class))))
+            })
+    void removeSecret(String id);
```

Comment:
We do not have any domain relation between `Assets`, `ContractAgreements` and secrets. If one should be introduced, this requires discussion in - and approval of - the technical committers and must be preceded by a decision-record. 

In addition, if we wanted to do that (which I do not), we would need to make sure that secrets are cleaned up when the parent entity, e.g. Assets, are deleted, etc.

Reply:
Hello @paullatzelsperger 
we discussed that item a few weeks ago during the committer meeting already. I explained the use-case and we concluded it was ok to have it in the codebase as long as this secret API was not embedded by default in the management API BOM which is the case here. 
If needed, I can write a decision-record tough.

Reply:
having the API is fine, and we discussed that. Introducing a domain relation between Assets, Contracts and Secrets is something different entirely, which - to my recollection - was not discussed. 
In other words, I object to the 409 here, not the Secrets API per se.